doc/dev/crimson: add instructions on how to build crimson (#37384)
ceph/ceph: doc/dev/crimson.rst

keep the support of HDDs and low-end SSDs via BlueStore. Crismon will try to
be backward compatible with classic OSD.

.. highlight:: console

Building Crimson
================

Crismon is not enabled by default. To enabled it::

  $ mkdir build && cd build
  $ cmake -DWITH_SEASTAR=ON ..

Please note, `ASan`_ is enabled by default if crimson is built from a source
cloned using git.

> Also, Seastar uses its own lockless allocator which does not play well with

Author: i searched "uses its own" and "its own" using google, and tried to find the evidences and/or counter evidences of "uses its own <adj>",  but it seems that people are using "its own <adj> <noun>" everywhere. to be more specific, probably you could try search "its own main processors".

the alien threads. So, to use alienstore / bluestore backend, you might want to
pass ``-DSeastar_CXX_FLAGS=-DSEASTAR_DEFAULT_ALLOCATOR`` to ``cmake`` when
configuring this project to use the libc allocator, like::

  $ cmake -DWITH_SEASTAR=ON -DSeastar_CXX_FLAGS=-DSEASTAR_DEFAULT_ALLOCATOR ..

.. _ASan: https://github.com/google/sanitizers/wiki/AddressSanitizer

Running Crimson
===============

As you might expect, crimson is not featurewise on par with its predecessor yet.

object store backend
--------------------

> ``crimson-osd`` now supports two object store backends:

Reviewer: ```suggestion
At the moment ``crimson-osd`` offers two object store backends:
```

``--osd-args``
    pass extra command line options to crimson-osd or ceph-osd. It's quite
    useful for passing Seastar options to crimson-osd. For instance, you could
    use ``--osd-args "--memory 2G"`` to set the memory to use. Please refer
    the output of::

      crimson-osd --help-seastar

    for more Seastar specific command line options.

``--memstore``

>     use the CyanStore as the objectstore backend.

Reviewer: ```suggestion
    use the CyanStore as the object store backend.
```

``--bluestore``

>     use the AlienStore as the objectstore backend. This is the default setting,

Reviewer: ```suggestion
    use the AlienStore as the object store backend. This is the default setting,
```

We should be consistent with the naming introduced by:

```
``crimson-osd`` now supports two object store backends:
```

Debugging Crimson
=================

Debugging with GDB
------------------

The `tips`_ for debugging Scylla also apply to Crimson.

.. _tips: https://github.com/scylladb/scylla/blob/master/docs/debugging.md#tips-and-tricks

> Decipher Backtrace with addr2line

Reviewer: How about:

```suggestion
Human-readable backtraces with addr2line
```
?

Author: i like the name. but i want to have symmetric titles for the both cases. i just removed the section title of "debugging crashes of teuthology based tests".
